cmd/libsnap-confine-private/mountinfo.c

```diff
@@ -206,27 +206,73 @@ static struct mountinfo_entry *parse_mountinfo_entry(const char *line)
 	// this extra memory to hold data parsed from the original line. In the
 	// end, the result is similar to using strtok except that the source and
 	// destination buffers are separate.
+	//
+	// At the end of the parsing process, the input buffer (line) and the
+	// output buffer (entry->line_buf) are the same except for where spaces
+	// were converted into NUL bytes (string terminators) and except for the
+	// leading part of the buffer that contains mount_id, parent_id, dev_major
+	// and dev_minor integer fields that are parsed separately.
+	//
+	// If MOUNTINFO_DEBUG is defined then extra debugging is printed to stderr
+	// and this allows for visual analysis of what is going on.
 	struct mountinfo_entry *entry =
 	    calloc(1, sizeof *entry + strlen(line) + 1);
 	if (entry == NULL) {
 		return NULL;
 	}
+#ifdef MOUNTINFO_DEBUG
+	// Poison the buffer with '\1' bytes that are printed as '#' characters
+	// by show_buffers() below. This is "unaltered" memory.
+	memset(entry->line_buf, 1, strlen(line));
+#endif				// MOUNTINFO_DEBUG
 	int nscanned;
-	int offset, total_offset = 0;
+	int offset_delta, offset = 0;
 	nscanned = sscanf(line, "%d %d %u:%u %n",
 			  &entry->mount_id, &entry->parent_id,
-			  &entry->dev_major, &entry->dev_minor, &offset);
+			  &entry->dev_major, &entry->dev_minor, &offset_delta);
 	if (nscanned != 4)
 		goto fail;
-	total_offset += offset;
-	int total_used = 0;
+	offset += offset_delta;
+
+	void show_buffers() {
+#ifdef MOUNTINFO_DEBUG
```

Comment:
I can call it whatever we want. I don't think we need this in production builds. I was about to remove this code but I think having it adds some insight into how this code works.